File: Valid Sudoku - Leetcode 36/Valid Sudoku - Leetcode 36.js
var isValidSudoku = function(board) {
    // Validate Rows
    for (let i = 0; i < 9; i++) {
        let set = new Set();
        for (let j = 0; j < 9; j++) {
            let item = board[i][j];
            if (item !== '.' && set.has(item)) {
                return false;
            }
            set.add(item);
        }
    }

    // Validate Columns
    for (let i = 0; i < 9; i++) {
        let set = new Set();
        for (let j = 0; j < 9; j++) {
            let item = board[j][i];
            if (item !== '.' && set.has(item)) {
                return false;
            }
            set.add(item);
        }
    }

    // Validate 3x3 Sub-grids
    let starts = [[0, 0], [0, 3], [0, 6],
                  [3, 0], [3, 3], [3, 6],
                  [6, 0], [6, 3], [6, 6]];
    
    for (let [startRow, startCol] of starts) {
        let set = new Set();
        for (let row = startRow; row < startRow + 3; row++) {
            for (let col = startCol; col < startCol + 3; col++) {
                let item = board[row][col];
                if (item !== '.' && set.has(item)) {
                    return false;
                }
                set.add(item);
            }
        }
    }

    return true;
};

File: Valid Sudoku - Leetcode 36/Valid Sudoku - Leetcode 36.cpp
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        // Validate Rows
        for (int i = 0; i < 9; i++) {
            unordered_set<char> set;
            for (int j = 0; j < 9; j++) {
                char item = board[i][j];
                if (item != '.' && !set.insert(item).second) {
                    return false;
                }
            }
        }

        // Validate Columns
        for (int i = 0; i < 9; i++) {
            unordered_set<char> set;
            for (int j = 0; j < 9; j++) {
                char item = board[j][i];
                if (item != '.' && !set.insert(item).second) {
                    return false;
                }
            }
        }

        // Validate 3x3 Sub-grids
        vector<vector<int>> starts = {{0, 0}, {0, 3}, {0, 6},
                                      {3, 0}, {3, 3}, {3, 6},
                                      {6, 0}, {6, 3}, {6, 6}};
        
        for (auto& start : starts) {
            unordered_set<char> set;
            for (int row = start[0]; row < start[0] + 3; row++) {
                for (int col = start[1]; col < start[1] + 3; col++) {
                    char item = board[row][col];
                    if (item != '.' && !set.insert(item).second) {
                        return false;
                    }
                }
            }
        }

        return true;
    }
};
